modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+            print(' Exiting...')
+            return False
+        waiter.wait(VolumeIds=current_volume_set)
+        set_index += set_count
+        if not running:
+            break
+    cleanup(client)
+    return True
+
+
+def modify_volume_set(client, print, func, instance_id, volume_id_set):
+    """Helper function to load volumes on an instance to not overload the
+    instance.
+
+    Args:
+        client (boto3.client): client to interact with AWS
+        print (func): Overwritten built-in print function
+        func (str): Function sname to modify_volume_set
+        instance_id (str): instance_id to attach volumes to
+        volume_ids (list): list of volume_ids to (de)attach to the instance.
+    Returns:
+        bool: True if the volumes were successfully attached.
+    """
+    for volume_id in volume_id_set:
+        try:
+            kwargs = {
+                'InstanceId':instance_id,
+                'VolumeId':volume_id
+            }
+            if func == 'attach_volume':
+                kwargs['Device'] = get_valid_device(client, instance_id)
+            caller = getattr(client, func)
+            caller(**kwargs)
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if  code == 'UnauthorizedOperation':
+                print('  FAILURE MISSING AWS PERMISSIONS')
+            else:
+                print(error)
+            return False
+    return True
+
+def get_valid_device(client, instance):
+    """Returns the next device mapping available
+
+    Args:
+        client (boto3.client): Client that gets the current block device mappings
+        instance (str): InstanceId to get curretn block device mappings
+    Returns:
+        str: Returns next mapping in form of /dev/xvd[base], otherwise /dev/xvdzz
+
+    """
+    response = client.describe_instances(InstanceIds=[instance])
+    mappings = response['Reservations'][0]['Instances'][0]['BlockDeviceMappings']
+    current_mappings = [device['DeviceName'] for device in mappings]
+    base_mappings = [char for char in 'bcdefghijklmnoqrstuvwxyz']
+    for base in base_mappings:
+        if '/dev/xvd' + base not in current_mappings:
+            return '/dev/xvd' + base
+    return '/dev/xvdzz'
+
+
+def get_snapshots(pacu, session, region):
+    """Returns snapshots given an AWS region
+    Args:
+        pacu (Main): Reference to Pacu
+        session (PacuSession): Reference to the Pacu session database
+        regions (list): Regions to check for snapshots
+    Returns:
+        dict: Mapping regions to corresponding list of snapshot_ids.
+    """
+    ec2_data = deepcopy(session.EC2)
+    if 'Snapshots' not in ec2_data:
+        fields = ['EC2', 'Snapshots']
+        module = 'enum_ebs_volumes_snapshots'
+        fetched_ec2_instances = pacu.fetch_data(fields, module)
+        if fetched_ec2_instances is False:
+            return None
+    snapshot_ids = []
+    for snapshot in ec2_data['Snapshots']:
```

Comment:
I think that if ec2_data exists but Snapshots isn't in it, ec2_data won't be updated by the fetch_data call because you deepcopied this dict before fetching the data and didn't replace it